pandas/io/parsers.py

```diff
@@ -1881,7 +1881,11 @@ def __init__(self, src: FilePathOrBuffer, **kwds):
             # no attribute "mmap"  [union-attr]
             self.handles.handle = self.handles.handle.mmap  # type: ignore[union-attr]
 
-        self._reader = parsers.TextReader(self.handles.handle, **kwds)
+        try:
+            self._reader = parsers.TextReader(self.handles.handle, **kwds)
+        except Exception as err:
```

Comment:
ahh here's the issue

```
try:
 .....
finally:
     self.handle.close()
```

should do it

Reply:
we *want* the original exception to raise as is, but of course close if needed.

Reply:
I did that first but unfortunately we might need the reader (and the file handles) if we iteratively read in chunks. We can only close the handle in case of an error. 

I think a plain `raise` would be the best solution? It doesn't seem to add anything new to the traceback.